### PR TITLE
SONARPY-57 Migrate to last SQ LTS 4.5.2

### PR DESCRIPTION
**Pull request by vilchik-elena**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -114,7 +114,7 @@
       <dependency>
         <groupId>org.codehaus.sonar.sslr-squid-bridge</groupId>
         <artifactId>sslr-squid-bridge</artifactId>
-        <version>2.4</version>
+        <version>2.5</version>
```
Comment (pynicolas) [inline pom.xml]: why not sslr-squid-bridge 2.5.3?

File: python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java [hunk 2]
```diff
@@ -28,8 +27,7 @@
 
 @Rule(
   key = "CommentRegularExpression",
-  priority = Priority.MAJOR,
-  cardinality = Cardinality.MULTIPLE)
```
Comment (pynicolas) [inline python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java]: Removing the cardinality makes the rule a simple rule instead of a rule template.

File: python-checks/src/main/java/org/sonar/python/checks/XPathCheck.java [hunk 3]
```diff
@@ -22,15 +22,13 @@
...
 import org.sonar.squidbridge.checks.AbstractXPathCheck;
-import org.sonar.check.Cardinality;
 import org.sonar.check.Priority;
 import org.sonar.check.Rule;
 import org.sonar.check.RuleProperty;
 
 @Rule(
   key = "XPath",
-  priority = Priority.MAJOR,
-  cardinality = Cardinality.MULTIPLE)
```
Comment (pynicolas) [inline python-checks/src/main/java/org/sonar/python/checks/XPathCheck.java]: Removing the cardinality makes the rule a simple rule instead of a rule template.

File: python-squid/src/test/java/org/sonar/python/FileLinesVisitorTest.java [hunk 4]
```diff
@@ -45,30 +43,25 @@
...
-  @Mock
-  private Project project;
-  @Mock
-  private FileLinesContextFactory fileLinesContextFactory;
-  @Mock
-  private ProjectFileSystem fileSystem;
-
   @Before
   public void before() {
     MockitoAnnotations.initMocks(this);
```
Comment (pynicolas) [inline python-squid/src/test/java/org/sonar/python/FileLinesVisitorTest.java]: MockitoAnnotations.initMocks() is not needed any more if no mock annotation is used.

File: sonar-python-plugin/pom.xml [hunk 5]
```diff
@@ -25,6 +25,7 @@
   <properties>
     <sonar.pluginClass>org.sonar.plugins.python.PythonPlugin</sonar.pluginClass>
     <sonar.pluginName>Python</sonar.pluginName>
+    <!--<sonar.skipDependenciesPackaging>true</sonar.skipDependenciesPackaging>-->
```
Comment (pynicolas) [inline sonar-python-plugin/pom.xml]: Please remove that line.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonReportSensor.java [hunk 6]
```diff
@@ -38,28 +36,29 @@
...
 
-  public PythonReportSensor(Settings conf, ModuleFileSystem fileSystem) {
+  public PythonReportSensor(Settings conf, FileSystem fileSystem) {
     this.conf = conf;
     this.fileSystem = fileSystem;
   }
 
   public boolean shouldExecuteOnProject(Project project) {
-    return !fileSystem.files(FileQuery.onSource().onLanguage(Python.KEY)).isEmpty();
+    return fileSystem.hasFiles(fileSystem.predicates().hasLanguage(Python.KEY));
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonReportSensor.java]: The new code is not equivalent to the old one: FileQuery.onSource() should be transformed in a predicate on file type.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java [hunk 7]
```diff
@@ -61,58 +59,57 @@
...
+        .addAnnotatedChecks(CheckList.getChecks());
     this.fileLinesContextFactory = fileLinesContextFactory;
     this.fileSystem = fileSystem;
-    this.resourcePerspectives = resourcePerspectives;
+    this.resourcePerspectives = perspectives;
   }
 
   public boolean shouldExecuteOnProject(Project project) {
-    return !fileSystem.files(FileQuery.onSource().onLanguage(Python.KEY)).isEmpty();
+    return fileSystem.hasFiles(fileSystem.predicates().hasLanguage(Python.KEY));
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java]: The new code is not equivalent to the old one: FileQuery.onSource() should be transformed in a predicate on file type.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java [hunk 8]
```diff
@@ -61,58 +59,57 @@
...
     this.context = context;
 
-    Collection<SquidAstVisitor<Grammar>> squidChecks = annotationCheckFactory.getChecks();
-    List<SquidAstVisitor<Grammar>> visitors = Lists.newArrayList(squidChecks);
-    visitors.add(new FileLinesVisitor(project, fileLinesContextFactory));
+    List<SquidAstVisitor<Grammar>> visitors = Lists.newArrayList(checks.all());
+    visitors.add(new FileLinesVisitor(fileLinesContextFactory, fileSystem));
     this.scanner = PythonAstScanner.create(createConfiguration(), visitors.toArray(new SquidAstVisitor[visitors.size()]));
-    scanner.scanFiles(fileSystem.files(FileQuery.onSource().onLanguage(Python.KEY)));
+    scanner.scanFiles(Lists.newArrayList(fileSystem.files(fileSystem.predicates().hasLanguage(Python.KEY))));
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java]: Same thing as above on file predicate.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java [hunk 9]
```diff
@@ -132,22 +129,22 @@ private void saveFunctionsComplexityDistribution(File sonarFile, SourceFile squi
...
     if (messages != null) {
       for (CheckMessage message : messages) {
-        ActiveRule rule = annotationCheckFactory.getActiveRule(message.getCheck());
+        RuleKey ruleKey = checks.ruleKey((SquidAstVisitor<Grammar>) message.getCheck());
         Issuable issuable = resourcePerspectives.as(Issuable.class, sonarFile);
 
         if (issuable != null) {
           Issue issue = issuable.newIssueBuilder()
-            .ruleKey(RuleKey.of(rule.getRepositoryKey(), rule.getRuleKey()))
+            .ruleKey(RuleKey.of(ruleKey.repository(), ruleKey.rule()))
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java]: Why do we need to create a new instance of RuleKey?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java [hunk 10]
```diff
@@ -104,124 +97,48 @@ public void analyse(Project project, SensorContext context) {
...
     LOG.debug("Parsing integration test coverage reports");
     List<File> itReports = getReports(conf, baseDir, IT_REPORT_PATH_KEY, IT_DEFAULT_REPORT_PATH);
     Map<String, CoverageMeasuresBuilder> itCoverageMeasures = parseReports(itReports);
-    saveMeasures(project, context, itCoverageMeasures, CoverageType.IT_COVERAGE);
+    saveMeasures(context, itCoverageMeasures, CoverageType.IT_COVERAGE);
 
     LOG.debug("Parsing overall test coverage reports");
     List<File> overallReports = getReports(conf, baseDir, OVERALL_REPORT_PATH_KEY, OVERALL_DEFAULT_REPORT_PATH);
     Map<String, CoverageMeasuresBuilder> overallCoverageMeasures = parseReports(overallReports);
-    saveMeasures(project, context, overallCoverageMeasures, CoverageType.OVERALL_COVERAGE);
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java]: Please undo this change.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java [hunk 11]
```diff
@@ -60,17 +59,19 @@ public PylintSensor(RuleFinder ruleFinder, PylintConfiguration conf, RulesProfil
   }
 
   public boolean shouldExecuteOnProject(Project project) {
-    return !fileSystem.files(FileQuery.onSource().onLanguage(Python.KEY)).isEmpty()
+    boolean x1 = fileSystem.hasFiles(fileSystem.predicates().hasLanguage(Python.KEY)),
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: x1 and x2 are unused.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java [hunk 12]
```diff
@@ -60,17 +59,19 @@ public PylintSensor(RuleFinder ruleFinder, PylintConfiguration conf, RulesProfil
...
+        x2 = !profile.getActiveRulesByRepository(PylintRuleRepository.REPOSITORY_KEY).isEmpty();
+    return fileSystem.hasFiles(fileSystem.predicates().hasLanguage(Python.KEY))
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: we miss the predicate on file type

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java [hunk 13]
```diff
@@ -60,17 +59,19 @@ public PylintSensor(RuleFinder ruleFinder, PylintConfiguration conf, RulesProfil
...
         && !profile.getActiveRulesByRepository(PylintRuleRepository.REPOSITORY_KEY).isEmpty();
   }
 
   public void analyse(Project project, SensorContext sensorContext) {
-    File workdir = new File(fileSystem.workingDir(), "/pylint/");
-    prepareWorkDir(workdir);
+    File workDir = new File(fileSystem.baseDir(), "/pylint/");
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: why not fileSystem.workDir()?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java [hunk 14]
```diff
@@ -60,17 +59,19 @@ public PylintSensor(RuleFinder ruleFinder, PylintConfiguration conf, RulesProfil
...
+    prepareWorkDir(workDir);
     int i = 0;
-    for (File file : fileSystem.files(FileQuery.onSource().onLanguage(Python.KEY))) {
+    for (File file : fileSystem.files(fileSystem.predicates().hasLanguage(Python.KEY))) {
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: The new code needs a predicate on file type.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/xunit/PythonXunitSensor.java [hunk 15]
```diff
@@ -87,52 +70,8 @@ protected String defaultReportPath() {
     return DEFAULT_REPORT_PATH;
   }
 
-  protected void processReports(final Project project, final SensorContext context, List<File> reports)
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/xunit/PythonXunitSensor.java]: Please undo this change.

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/coverage/PythonCoverageSensorTest.java [hunk 16]
```diff
@@ -19,55 +19,62 @@
...
+    fs.add(new DefaultInputFile("sources/file1.py"));
+    fs.add(new DefaultInputFile("sources/file2.py"));
+    fs.add(new DefaultInputFile("sources/file3.py"));
+    fs.add(new DefaultInputFile("sources/file4.py"));
+    fs.add(new DefaultInputFile("sources/file5.py"));
+    fs.add(new DefaultInputFile("sources/file6.py"));
+    fs.add(new DefaultInputFile("sources/file7.py"));
     sensor.analyse(project, context);
-    verify(context, times(66)).saveMeasure((Resource) anyObject(), any(Measure.class));
+    verify(context, times(33)).saveMeasure((InputFile) anyObject(), any(Measure.class));
```
Comment (pynicolas) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/coverage/PythonCoverageSensorTest.java]: did we change a behavior?

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java [hunk 17]
```diff
@@ -67,37 +61,29 @@ public void shouldExecuteOnlyWhenNecessary() {
     // which means: only on python projects and only if
     // there is at least one active pylint rule
 
-    Project pythonProject = createProjectForLanguage(Python.KEY);
-    Project foreignProject = createProjectForLanguage("whatever");
+    DefaultFileSystem fileSystemPython = new DefaultFileSystem();
+    DefaultFileSystem fileSystemForeign = new DefaultFileSystem();
+    DefaultInputFile inputFile1 = new DefaultInputFile("src/test/resources/example_project/example.swift").setLanguage(Python.KEY);
```
Comment (pynicolas) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java]: strange example...

File: pom.xml (hunk 1 above)
Comment (vilchik-elena) [inline pom.xml]: changed back on 2.4 in c0b169360671650bfba43406823e330bb6cf0007

File: python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java (hunk 2 above)
Comment (vilchik-elena) [inline python-checks/src/main/java/org/sonar/python/checks/CommentRegularExpressionCheck.java]: Changed back in c0b169360671650bfba43406823e330bb6cf0007

File: python-checks/src/main/java/org/sonar/python/checks/XPathCheck.java (hunk 3 above)
Comment (vilchik-elena) [inline python-checks/src/main/java/org/sonar/python/checks/XPathCheck.java]: Changed back in c0b169360671650bfba43406823e330bb6cf0007

File: python-squid/src/test/java/org/sonar/python/FileLinesVisitorTest.java (hunk 4 above)
Comment (vilchik-elena) [inline python-squid/src/test/java/org/sonar/python/FileLinesVisitorTest.java]: initMocks() removed in c0b169360671650bfba43406823e330bb6cf0007

File: sonar-python-plugin/pom.xml (hunk 5 above)
Comment (vilchik-elena) [inline sonar-python-plugin/pom.xml]: Line removed in c0b169360671650bfba43406823e330bb6cf0007

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonReportSensor.java (hunk 6 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonReportSensor.java]: Predicate on file type added in c0b169360671650bfba43406823e330bb6cf0007

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java (hunk 7 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java]: Predicate on file type added in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java (hunk 8 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java]: Predicate on file type added in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java (hunk 9 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSquidSensor.java]: Old instance of ruleKey is used c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java (hunk 10 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/coverage/PythonCoverageSensor.java]: Changed back in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java (hunk 11 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: x1 and x2 are renamed and used in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java (hunk 12 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: Predicate on file type added in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java (hunk 13 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: changed in workDir() in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java (hunk 14 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java]: Predicate on file type added in c0b1693

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/xunit/PythonXunitSensor.java (hunk 15 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/xunit/PythonXunitSensor.java]: Changed back in c0b1693

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/coverage/PythonCoverageSensorTest.java (hunk 16 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/coverage/PythonCoverageSensorTest.java]: Good question, need some investigation. I will try to figure it out.

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java (hunk 17 above)
Comment (vilchik-elena) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java]: sorry:) i changed file extension in c0b1693
